x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlLogoutResponseHandlerHttpRedirectTests.java

```diff
@@ -0,0 +1,132 @@
+        issuer.setValue(IDP_ENTITY_ID);
+        logoutResponse.setIssuer(issuer);
+        final String url = new SamlRedirect(logoutResponse, signingConfiguration).getRedirectUrl();
+        samlLogoutResponseHandler.handle(new URI(url).getRawQuery(), List.of(requestId));
+    }
+
+    public void testHandlerFailsIfStatusIsNotSuccess() {
+        final String requestId = SamlUtils.generateSecureNCName(randomIntBetween(8, 30));
+        final SigningConfiguration signingConfiguration = new SigningConfiguration(Sets.newHashSet("*"), credential);
+        final LogoutResponse logoutResponse = SamlUtils.buildObject(LogoutResponse.class, LogoutResponse.DEFAULT_ELEMENT_NAME);
+        logoutResponse.setDestination(LOGOUT_URL);
+        logoutResponse.setIssueInstant(new DateTime(clock.millis()));
+        logoutResponse.setID(SamlUtils.generateSecureNCName(randomIntBetween(8, 30)));
+        logoutResponse.setInResponseTo(requestId);
+        logoutResponse.setStatus(buildStatus(randomFrom(StatusCode.REQUESTER, StatusCode.RESPONDER)));
+
+        final Issuer issuer = SamlUtils.buildObject(Issuer.class, Issuer.DEFAULT_ELEMENT_NAME);
+        issuer.setValue(IDP_ENTITY_ID);
+        logoutResponse.setIssuer(issuer);
+        final String url = new SamlRedirect(logoutResponse, signingConfiguration).getRedirectUrl();
+
+        final ElasticsearchSecurityException e =
+            expectSamlException(() -> samlLogoutResponseHandler.handle(new URI(url).getRawQuery(), List.of(requestId)));
+        assertThat(e.getMessage(), containsString("is not a 'success' response"));
+    }
+
+    public void testHandlerWillUseHttpPostBindingWhenUrlNotSigned() {
+        final String requestId = SamlUtils.generateSecureNCName(randomIntBetween(8, 30));
+        final SigningConfiguration signingConfiguration = new SigningConfiguration(Sets.newHashSet("*"), null);
+        final LogoutResponse logoutResponse = SamlUtils.buildObject(LogoutResponse.class, LogoutResponse.DEFAULT_ELEMENT_NAME);
+        logoutResponse.setDestination(LOGOUT_URL);
+        logoutResponse.setIssueInstant(new DateTime(clock.millis()));
+        logoutResponse.setID(SamlUtils.generateSecureNCName(randomIntBetween(8, 30)));
+        logoutResponse.setInResponseTo(requestId);
+        logoutResponse.setStatus(buildStatus(randomFrom(StatusCode.REQUESTER, StatusCode.RESPONDER)));
+
+        final Issuer issuer = SamlUtils.buildObject(Issuer.class, Issuer.DEFAULT_ELEMENT_NAME);
+        issuer.setValue(IDP_ENTITY_ID);
+        logoutResponse.setIssuer(issuer);
+        final String url = new SamlRedirect(logoutResponse, signingConfiguration).getRedirectUrl();
+        final ElasticsearchSecurityException e =
+            expectSamlException(() -> samlLogoutResponseHandler.handle(new URI(url).getRawQuery(), List.of(requestId)));
+        assertThat(e.getMessage(), containsString("Failed to parse SAML message"));
```

Comment:
related to the discussion above, this is a case where we don't want to fail to parse the SAML message because we will try to parse it as if it came from the HTTP POST binding. We want to fail this because it is an unsigned request that came via the HTTP-Redirect binding and we disallow that.

Reply:
I think I see your point here. I'll move away from auto-detect and have two separate fields at request level.